File: aspnet-in-practice/15/VB/VirtualPathProvider/App_Code/Utility.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Configuration
Imports VirtualFileSystemModel

Namespace ASPNET4InPractice
	Friend Module Utility
		Friend ReadOnly Property BasePath() As String
			Get
				Return If(ConfigurationManager.AppSettings("virtualDirectory"), "~/")
			End Get
		End Property

		Friend Function FileExists(ByVal virtualPath As String) As Boolean
			Using ctx As New VirtualFileSystemEntities()
				Return ctx.VirtualFiles.Count(Function(x) x.VirtualPath.Equals(virtualPath)) > 0
			End Using
		End Function

		Friend Function GetFileContents(ByVal virtualPath As String) As String
			Using ctx As New VirtualFileSystemEntities()
				Return ctx.VirtualFiles.First(Function(x) x.VirtualPath.Equals(virtualPath)).Contents
			End Using
		End Function

		Friend Function GetLastModifiedTimeStamp(ByVal virtualPath As String) As Byte()
			Using ctx As New VirtualFileSystemEntities()
				Return ctx.VirtualFiles.First(Function(x) x.VirtualPath.Equals(virtualPath)).LastUpdated
			End Using
		End Function
	End Module
End Namespace

File: aspnet-in-practice/15/CS/VirtualPathProvider/App_Code/Utility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;
using VirtualFileSystemModel;

namespace ASPNET4InPractice
{
	internal static class Utility
	{
		internal static string BasePath
		{
			get
			{
				return ConfigurationManager.AppSettings["virtualDirectory"]??"~/";
			}
		}

		internal static bool FileExists(string virtualPath)
		{
			using (VirtualFileSystemEntities ctx = new VirtualFileSystemEntities())
			{
				return ctx.VirtualFiles.Count(x => x.VirtualPath.Equals(virtualPath)) > 0;
			}
		}

		internal static string GetFileContents(string virtualPath)
		{
			using (VirtualFileSystemEntities ctx = new VirtualFileSystemEntities())
			{
				return ctx.VirtualFiles.First(x => x.VirtualPath.Equals(virtualPath)).Contents;
			}
		}

		internal static byte[] GetLastModifiedTimeStamp(string virtualPath)
		{
			using (VirtualFileSystemEntities ctx = new VirtualFileSystemEntities())
			{
				return ctx.VirtualFiles.First(x => x.VirtualPath.Equals(virtualPath)).LastUpdated;
			}
		}
	}
}